Stripe/STPEphemeralKeyManager.m

```diff
@@ -64,18 +68,44 @@ - (void)handleWillForegroundNotification {
     // eager refreshses to once per hour.
     if (!self.currentKeyIsUnexpired && self.shouldPerformEagerRefresh) {
         self.lastEagerKeyRefresh = [NSDate date];
-        [self.keyProvider createCustomerKeyWithAPIVersion:self.apiVersion completion:^(NSDictionary *jsonResponse, __unused NSError *error) {
-            STPEphemeralKey *key = [STPEphemeralKey decodedObjectFromAPIResponse:jsonResponse];
-            if (key) {
-                self.customerKey = key;
-            }
+        [self getOrCreateKey:^(__unused STPEphemeralKey * _Nullable ephemeralKey, __unused NSError * _Nullable error) {
+            
         }];
     }
 }
 
-- (void)getCustomerKey:(STPEphemeralKeyCompletionBlock)completion {
+- (void)createKey {
+    STPJSONResponseCompletionBlock jsonCompletion = ^(NSDictionary *jsonResponse, NSError *error) {
+        STPEphemeralKey *key = [STPEphemeralKey decodedObjectFromAPIResponse:jsonResponse];
+        if (key) {
+            [self.createKeyPromise succeed:key];
+        } else {
+            // the API request failed
+            if (error) {
+                [self.createKeyPromise fail:error];
+            }
+            // the ephemeral key could not be decoded
+            else {
+                [self.createKeyPromise fail:[NSError stp_ephemeralKeyDecodingError]];
+                NSAssert(NO, @"Could not parse the ephemeral key response. Make sure your backend is sending the unmodified JSON of the ephemeral key to your app. For more info, see https://stripe.com/docs/mobile/ios/standard#prepare-your-api");
```

Comment:
It might be handy to log the type of key here (Issuing vs Customer), not sure if that's available in this context. My thought is the user will contact us when they see that error, and it'd be nice to know which key provider is causing the issue.

Reply:
👍